### PR TITLE
[#69] Convert loops

### LINKED ISSUE #69: Convert loops
`arr.forEach()` is used a lot, however, if we switch to `for (let i = 0; i < things.length; i++) {}` or `for (let thing of things) {}` then users have greater control, like a hook being able to add/remove additional tasks mid-run. This would allow for more dynamic build options/tooling.

### PR DESCRIPTION
**Pull request by aslamdoctor**
Convert forEach loops into for loop.
Fixes #69

Coverage Badge

**Related Issue:** #69

**Notes for reviewer:**

*

**Jared's Checklist:**

* [ ] Bump
* [ ] Update deps

### CODE REVIEW

**Comment by TheJaredWilcurt** [comment]
Thanks for the PR. Make sure the linter passes. Un-do any other changes unrelated to the PR (like breaking up code onto multiple lines), and remove the Package-lock downgrade, it was already on version 2, no need to go down to version 1 in a simple PR to replace `forEach`.

**Comment by aslamdoctor** [comment]
Oops, that was prettier extension in my vscode. Just pushed the linting-tested version.

File: src/validator.js [hunk 1]
```diff
@@ -446,14 +446,15 @@ const validator = {
...
+    for (let i = 0; i < documentedHooks.length; i++) {
+      let hook = documentedHooks[i];
       const hookType = typeof(hooksContainer[hook]);
       const allowedTypes = ['function', 'undefined'];
       if (!allowedTypes.includes(hookType)) {
         helpers.throwError(options, 'The ' + location + hook + ' must be a function or undefined.');
         delete hooksContainer[hook];
       }
-    });
+    }
```
Comment (TheJaredWilcurt) [inline src/validator.js:457]: There are 4 `forEach` in this file, 3 still need converted.

Comment (aslamdoctor) [reply src/validator.js:457]: Yes those 3 has static array data e.g `['styles', 'markup', 'scripts']` 
should I store it to a variable and then make it `for` loop?

Comment (aslamdoctor) [reply src/validator.js:464]: Nevermind, I have made the required changes to those 3 loops aswell 👍

Comment (TheJaredWilcurt) [reply src/validator.js:464]: Oh yeah, you're right, we don't really need those to be traditional for loops. I was just doing a `ctrl+f` across the code and didn't check that. Probably cleaner to revert them, like you had it originally, then we can merge this

Comment (aslamdoctor) [reply src/validator.js:464]: OK great, reverting back

**Comment by TheJaredWilcurt** [comment]
Thanks so much!

**Pull request merged**